src/m365/planner/commands/plan/plan-set.ts

```diff
@@ -0,0 +1,156 @@
+import { Cli, CommandOutput, Logger } from '../../../../cli';
+import Command, {
+  CommandOption,
+  CommandError
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import Utils from '../../../../Utils';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import commands from '../../commands';
+import request from '../../../../request';
+import { Options as PlanGetCommandOptions } from '../plan/plan-get';
+import * as planGetCommand from '../plan/plan-get';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  title?: string;
+  newTitle: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class PlannerPlanSetCommand extends GraphItemsListCommand<any> {
+  public get name(): string {
+    return commands.PLAN_SET;
+  }
+
+  public get description(): string {
+    return 'Set a Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (error?: any) => void): void {
+    const getPlan = (): Promise<CommandOutput> => {
+      if (this.verbose) {
+        logger.logToStderr(`Retrieving plan ${args.options.id || args.options.title} ...`);
+      }
+
+      const options: PlanGetCommandOptions = {
+        ...args.options,
+        output: 'json',
+        debug: this.debug,
+        verbose: this.verbose
+      };
+
+      return Cli.executeCommandWithOutput(planGetCommand as Command, { options: { ...options, _: [] } });
+    };
+
+    const updatePlan = (plan: any): Promise<void> => {
+      if (this.verbose) {
+        logger.logToStderr(`Updating plan with id ${plan['id']} ...`);
```

Comment:
When we change `plan: any` to a specific type, let's update this line to use the ID property instead of indexer so that we can get design-time type checking